config/defines.inc.php

```diff
@@ -24,9 +24,15 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
+$debugIps = require __DIR__.DIRECTORY_SEPARATOR.'debug_ip.php';
+
 /* Debug only */
 if (!defined('_PS_MODE_DEV_')) {
-    define('_PS_MODE_DEV_', true);
+    if (in_array($_SERVER['REMOTE_ADDR'], $debugIps)) {
+        define($debugConfigName = '_PS_MODE_DEV_', true);
```

Comment:
When switching debug mode on/off, it looks for regex pattern in file to rewrite it's value, using variable does the trick to avoid switching value on this line.